hilti/toolchain/src/ast/node.cc

```diff
@@ -23,7 +23,10 @@ std::string node::to_string(const Tags& ti) {
     return util::join(util::transform(ti, [](auto i) { return std::to_string(i); }), ",");
 }
 
-Node::~Node() {}
+Node::~Node() {
+    clearChildren();
```

Comment:
Hmm, I guess this manual lifetime management wouldn't be needed if the children where `PinnedNodePtr` instead of simply `Nodes`. Is that impossible since it would break the use case of temporarily having unpinned nodes on the AST?

Reply:
No, different reason: we have a few places where we expect `_children` to be a vector of `Node*`, so that we can access children, or ranges of children, to retrieve raw node pointers directly without having to dereference/convert them first for further usage elsewhere (we use raw pointers almost everywhere else).